File: Abstraction/Pattern Matching/area.hs
printArea :: [Int] -> IO ()
printArea [side] = print $ "Area of square is: " ++ show (side * side)
printArea [height, width] = print $ "Area of rectangle is: " ++ show (height * width)

main :: IO ()
main = do
  printArea [5]
  printArea [4, 6]

File: Abstraction/Pattern Matching/area.fs
open System

let printArea = function
    | side -> printfn "Area of square is: %d" (side * side)
    | height, width -> printfn "Area of rectangle is: %d" (height * width)

[<EntryPoint>]
let main _ =
    printArea 5
    printArea (4, 6)
    0
